eventqueue/processor_test.go

```diff
@@ -118,9 +100,11 @@ func TestProcessor(t *testing.T) {
 			require.NoError(t, err)
 		}
 
-		// Advance tickers by 1500ms to trigger the first item
+		assert.Eventually(t, clock.HasWaiters, time.Second, 100*time.Millisecond)
+
+		// Advance tickers by 1s to trigger the first item
 		t.Log("Waiting for signal 1")
-		advanceTickers(500*time.Millisecond, 3)
```

Comment:
I made it advance in steps before also to assert that things do need 3 ticks to trigger. That was my thought at least.

Reply:
What is the thinking behind needing 3 distinct ticks, rather than a single larger tick?